pallets/cash/src/internal/events.rs

```diff
@@ -102,29 +102,41 @@ pub fn track_chain_events_on<T: Config>(chain_id: ChainId) -> Result<(), Reason>
     let starport = get_starport::<T>(chain_id)?;
     let me = get_current_validator::<T>()?;
     let last_block = get_last_block::<T>(chain_id)?;
-    let true_block = fetch_chain_block(chain_id, last_block.number(), starport)?;
-    if last_block.hash() == true_block.hash() {
+    let next_block_number = last_block
+        .number()
+        .checked_add(1)
+        .ok_or(MathError::Overflow)?;
+    let next_block = fetch_chain_block(chain_id, next_block_number, starport)?;
+    if last_block.hash() == next_block.parent_hash() {
         debug!(
-            "Worker sees the same fork: true={:?} last={:?}",
-            true_block, last_block
+            "Worker sees the same fork: next={:?} last={:?}",
+            next_block, last_block
         );
         let pending_blocks = PendingChainBlocks::get(chain_id);
         let event_queue = get_event_queue::<T>(chain_id)?;
         let slack = queue_slack(&event_queue) as u64;
-        let blocks = fetch_chain_blocks(
-            chain_id,
-            last_block.number() + 1,
-            last_block.number() + 1 + slack,
-            starport,
-        )?
-        .filter_already_supported(&me.substrate_id, pending_blocks);
+        let blocks = next_block.concat(
```

Comment:
feels like we should at least verify the parent_hash chain when fetching here, similar to the way re-orgs do. Am I missing something?

Reply:
The reorgs don't verify, the chain is ultimately checked by the runtime